File: backend/src/main/java/org/akashihi/mdg/dao/BudgetEntryRepository.java
package org.akashihi.mdg.dao;

import org.akashihi.mdg.entity.Budget;
import org.akashihi.mdg.entity.BudgetEntry;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Collection;

@Repository
public interface BudgetEntryRepository extends JpaRepository<BudgetEntry, Long> {
    Collection<BudgetEntry> findByBudget(Budget b);
}

File: backend/src/main/java/org/akashihi/mdg/dao/BudgetEntryRepository.kt
package org.akashihi.mdg.dao

import org.akashihi.mdg.entity.Budget
import org.akashihi.mdg.entity.BudgetEntry
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BudgetEntryRepository : JpaRepository<BudgetEntry, Long> {
    fun findByBudget(b: Budget): Collection<BudgetEntry>
}